src/tables/colr.rs

```diff
@@ -49,13 +51,325 @@ impl FromData for LayerRecord {
     }
 }
 
-/// A trait for color glyph painting.
-pub trait Painter {
+/// A [BaseGlyphPaintRecord](
+/// https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug)]
+struct BaseGlyphPaintRecord {
+    glyph_id: GlyphId,
+    paint_table_offset: Offset32,
+}
+
+impl FromData for BaseGlyphPaintRecord {
+    const SIZE: usize = 6;
+
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(Self {
+            glyph_id: s.read::<GlyphId>()?,
+            paint_table_offset: s.read::<Offset32>()?,
+        })
+    }
+}
+
+/// A [gradient extend](
+/// https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug)]
+pub enum GradientExtend {
+    Pad,
+    Repeat,
+    Reflect,
+}
+
+impl FromData for GradientExtend {
+    const SIZE: usize = 1;
+
+    fn parse(data: &[u8]) -> Option<Self> {
+        match data[0] {
```

Comment:
won't panic in practice due to where it's called I assume, but still an unfamiliar sight in this crate since it can technically panic. any reason to prefer this over `data.first()?`?

Reply:
It's pretty common. `data` is guarantee to be `Self::SIZE`.

Reply:
Okay, good to know.